### ISSUE TITLE
longevity: Segfault in destroy_data_then_repair/StartStopService nemesis

### ISSUE THREAD

**Issue by amoskong** [scylladb/scylla#3063]
*Installation details*
Scylla version (or git commit hash): 2.1.rc0-0.20171210.be5127388
Cluster size: 6 db nodes, 4 loaders
OS (RHEL/CentOS/Ubuntu/AWS AMI): CentOS

Job: http://jenkins.cloudius-systems.com:8080/job/scylla-master-longevity-gce/label=gce3-qavpc/23/
Log backup: https://s3.amazonaws.com/scylladb-users-upload/scylla-qa-team/job-2017-12-12T23.27-c58f5c3.tar.gz

Description:
In 2.1 longevity test, #2820 occurred `4` times after `nodetool_drain` nemesis at the beginning. In following nemesis tests, some nodes were stop and failed to start for segfault error.

Background:
```
00:18:21,942 >>>Started random_disrupt_method nodetool_drain
00:22:35,072 <<<Finished random_disrupt_method nodetool_drain
...
00:24:32,506  | Skipping schema refresh due to lack of schema agreement
00:24:42,543  | Skipping schema refresh due to lack of schema agreement
00:24:52,566  | Skipping schema refresh due to lack of schema agreement
...
00:25:02,288 thread L0055 DEBUG| [control connection] Schemas mismatched, trying again
00:25:02,496 thread L0055 DEBUG| [control connection] Schemas mismatched, trying again
00:25:02,697 thread L0055 WARNI| Node 35.185.83.27 is reporting a schema disagreement: {UUID('8dc84a5a-26bd-38c8-9c19-4d60a9f405e3'): ['10.142.0.10', '10.142.0.11', '10.142.0.14', '35.185.83.27', '10.142.0.15'], UUID('6922d476-7eeb-35f6-a20d-8791c08936bd'): ['10.142.0.16']}
00:25:02,697  | Skipping schema refresh due to lack of schema agreement
...
00:29:19,212 >>>Started random_disrupt_method modify_table_comment
00:29:24,643  <<<<Finished random_disrupt_method modify_table_comment

00:39:26,035 [10.142.0.16] [stdout] Dec 13 00:39:18 err    |  [shard 8] view - Error applying view update to 10.142.0.15: exceptions::mutation_write_timeout_exception (Operation timed out for mview.users_by_last_name - received only 0 responses from 1 CL=ONE.)

00:36:11,578 >>>Started random_disrupt_method destroy_data_then_repair

```
Segfault err in starting node of destroy_data_then_repair nemesis:
```
[10.142.0.15] [stdout] Dec 13 00:37:39 info   | Segmentation fault on shard 4.
[10.142.0.15] [stdout] Dec 13 00:37:39 info   | Backtrace:
[10.142.0.15] [stdout] Dec 13 00:37:39 info   | 0x00000000004a376b
[10.142.0.15] [stdout] Dec 13 00:37:39 info   | 0x00000000004a383c
[10.142.0.15] [stdout] Dec 13 00:37:39 info   | 0x00000000004a3893
[10.142.0.15] [stdout] Dec 13 00:37:39 info   | /lib64/libpthread.so.0+0x000000000000f5df
[10.142.0.15] [stdout] Dec 13 00:37:39 info   | 0x000000000115dbc0
[10.142.0.15] [stdout] Dec 13 00:37:39 info   | 0x00000000004a012a
[10.142.0.15] [stdout] Dec 13 00:37:39 info   | 0x00000000004a040c
[10.142.0.15] [stdout] Dec 13 00:37:39 info   | 0x00000000004ef304
[10.142.0.15] [stdout] Dec 13 00:37:39 info   | 0x00000000004f49f2
[10.142.0.15] [stdout] Dec 13 00:37:39 info   | 0x00000000005576ad
[10.142.0.15] [stdout] Dec 13 00:37:39 info   | /lib64/libpthread.so.0+0x0000000000007e24
[10.142.0.15] [stdout] Dec 13 00:37:39 info   | /lib64/libc.so.6+0x00000000000f834c
[10.142.0.15] [stdout] Dec 13 00:39:25 notice | scylla-server.service: main process exited, code=killed, status=11/SEGV
[10.142.0.15] [stdout] Dec 13 00:39:25 err    | Failed to start Scylla Server.
[10.142.0.15] [stdout] Dec 13 00:39:25 err    | Dependency failed for Scylla JMX.
[10.142.0.15] [stdout] Dec 13 00:39:25 notice | Job scylla-jmx.service/start failed with result 'dependency'.
[10.142.0.15] [stdout] Dec 13 00:39:25 notice | Unit scylla-server.service entered failed state.
[10.142.0.15] [stdout] Dec 13 00:39:25 warning| scylla-server.service failed.
[10.142.0.15] [stdout] Dec 13 00:39:26 info   | scylla-server.service holdoff time over, scheduling restart.

00:46:04,619 >>Started random_disrupt_method nodetool_refresh
00:46:26,711 <<<Finished random_disrupt_method nodetool_refresh

00:53:05,779 >>>Started random_disrupt_method modify_table_gc_grace_time
2017-12-13 00:53:34,574 remote           L0197 DEBUG| [10.142.0.10] [stdout] <stdin>:1:Warning: schema version mismatch detected; check the schema versions of your nodes in system.local and system.peers.
```

Segfault occurred in another db node in second StopStartService nemesis:
```
01:00:16,425 >>>>Started random_disrupt_method stop_start_scylla_server
...
[10.142.0.14] [stdout] Dec 13 01:01:55 info   | Segmentation fault on shard 4.
[10.142.0.14] [stdout] Dec 13 01:01:55 info   | Backtrace: 
[10.142.0.14] [stdout] Dec 13 01:01:55 info   | 0x00000000004a376b
[10.142.0.14] [stdout] Dec 13 01:01:55 info   | 0x00000000004a383c
[10.142.0.14] [stdout] Dec 13 01:01:55 info   | 0x00000000004a3893
[10.142.0.14] [stdout] Dec 13 01:01:55 info   | /lib64/libpthread.so.0+0x000000000000f5df
[10.142.0.14] [stdout] Dec 13 01:01:55 info   | 0x000000000115dbc0
[10.142.0.14] [stdout] Dec 13 01:01:55 info   | 0x00000000004a012a
[10.142.0.14] [stdout] Dec 13 01:01:55 info   | 0x00000000004a040c
[10.142.0.14] [stdout] Dec 13 01:01:55 info   | 0x00000000004ef304
[10.142.0.14] [stdout] Dec 13 01:01:55 info   | 0x00000000004f49f2
[10.142.0.14] [stdout] Dec 13 01:01:55 info   | 0x00000000005576ad
[10.142.0.14] [stdout] Dec 13 01:01:55 info   | /lib64/libpthread.so.0+0x0000000000007e24
[10.142.0.14] [stdout] Dec 13 01:01:55 info   | /lib64/libc.so.6+0x00000000000f834c
[10.142.0.14] [stdout] Dec 13 01:03:39 notice | scylla-server.service: main process exited, code=killed, status=11/SEGV
[10.142.0.14] [stdout] Dec 13 01:03:39 err    | Failed to start Scylla Server.
[10.142.0.14] [stdout] Dec 13 01:03:39 err    | Dependency failed for Scylla JMX.
```

@tgrabiec @roydahan @slivne

**Comment by amoskong** [comment]
```
[scylla-test@longevity-50gb-4d-master-db-node-568b6ef3-0-4 scripts]$ ./seastar-addr2line -e /usr/bin/scylla
0x00000000004a376b
0x00000000004a383c
0x00000000004a3893
/lib64/libpthread.so.0+0x000000000000f5df
0x000000000115dbc0
0x00000000004a012a
0x00000000004a040c
0x00000000004ef304
0x00000000004f49f2
0x00000000005576ad
/lib64/libpthread.so.0+0x0000000000007e24
/lib64/libc.so.6+0x00000000000f834c
seastar::backtrace_buffer::append_backtrace() at /usr/src/debug/scylla-2.1.rc0/seastar/core/reactor.cc:294
 (inlined by) print_with_backtrace at /usr/src/debug/scylla-2.1.rc0/seastar/core/reactor.cc:315
_Unwind_RaiseException at /usr/src/debug/scylla-2.1.rc0/seastar/core/reactor.cc:322
_Unwind_RaiseException at /usr/src/debug/scylla-2.1.rc0/seastar/core/reactor.cc:3726
 (inlined by) operator() at /usr/src/debug/scylla-2.1.rc0/seastar/core/reactor.cc:3712
 (inlined by) _FUN at /usr/src/debug/scylla-2.1.rc0/seastar/core/reactor.cc:3708
_L_unlock_13 at funlockfile.c:?
seastar::lw_shared_ptr<std::vector<view_ptr, std::allocator<view_ptr> > >::operator*() const at /usr/src/debug/scylla-2.1.rc0/seastar/core/shared_ptr.hh:305
 (inlined by) operator() at /usr/src/debug/scylla-2.1.rc0/db/schema_tables.cc:2218
 (inlined by) apply at /usr/src/debug/scylla-2.1.rc0/seastar/core/apply.hh:36
 (inlined by) apply<db::schema_tables::create_views_from_schema_partition(seastar::distributed<service::storage_proxy>&, const mapped_type&)::<lambda()> > at /usr/src/debug/scylla-2.1.rc0/seastar/core/apply.hh:44
 (inlined by) apply<db::schema_tables::create_views_from_schema_partition(seastar::distributed<service::storage_proxy>&, const mapped_type&)::<lambda()> > at /usr/src/debug/scylla-2.1.rc0/seastar/core/future.hh:1232
 (inlined by) operator()<seastar::future_state<> > at /usr/src/debug/scylla-2.1.rc0/seastar/core/future.hh:905
 (inlined by) run at /usr/src/debug/scylla-2.1.rc0/seastar/core/future.hh:396
seastar::reactor::run_tasks(seastar::reactor::task_queue&) at /usr/src/debug/scylla-2.1.rc0/seastar/core/reactor.cc:2431
seastar::reactor::run_some_tasks(std::chrono::time_point<std::chrono::_V2::steady_clock, std::chrono::duration<long, std::ratio<1l, 1000000000l> > >&) at /usr/src/debug/scylla-2.1.rc0/seastar/core/reactor.cc:2841
seastar::reactor::run_some_tasks(std::chrono::time_point<std::chrono::_V2::steady_clock, std::chrono::duration<long, std::ratio<1l, 1000000000l> > >&) at /opt/scylladb/include/c++/7/bits/unique_ptr.h:267
 (inlined by) seastar::reactor::run() at /usr/src/debug/scylla-2.1.rc0/seastar/core/reactor.cc:2985
_Unwind_RaiseException at /usr/src/debug/scylla-2.1.rc0/seastar/core/reactor.cc:3931 (discriminator 3)
std::function<void ()>::operator()() const at /opt/scylladb/include/c++/7/bits/std_function.h:706
 (inlined by) seastar::posix_thread::start_routine(void*) at /usr/src/debug/scylla-2.1.rc0/seastar/core/posix.cc:52
start_thread at pthread_create.c:?
```

**Comment by amoskong** [comment]
### Workload info:

prepare_write_cmd: 
- "cassandra-stress write cl=QUORUM n=100000000 -schema 'replication(factor=3) compaction(strategy=SizeTieredCompactionStrategy)' -port jmx=6868 -mode cql3 native -rate threads=500 -pop seq=1..100000000"

stress_cmd: 
- "cassandra-stress mixed cl=QUORUM duration=5760m -schema 'replication(factor=3) compaction(strategy=SizeTieredCompactionStrategy)' -port jmx=6868 -mode cql3 native -rate threads=20 -pop seq=1..100000000 -log interval=5",
- "cassandra-stress counter_write cl=QUORUM duration=5760m -schema 'replication(factor=3) compaction(strategy=DateTieredCompactionStrategy)' -port jmx=6868 -mode cql3 native -rate threads=1 -pop seq=1..1000000",
- "cassandra-stress user profile=/tmp/cs_mv_profile.yaml ops'(insert=3,read1=1,read2=1,read3=1)' cl=QUORUM duration=5760m -port jmx=6868 -mode cql3 native -rate threads=10"

stress_read_cmd:
- "cassandra-stress read duration=5760m -port jmx=6868 -mode cql3 native -rate threads=20 -pop seq=1..100000000 -log interval=5",
- "cassandra-stress counter_read duration=5760m -port jmx=6868 -mode cql3 native -rate threads=10 -pop seq=1..1000000"

**Comment by duarten** [comment]
Something's off here. The code the backtrace is pointing to is:

```c++
future<std::vector<view_ptr>> create_views_from_schema_partition(distributed<service::storage_proxy>& proxy, const schema_result::mapped_type& result)
{
    return do_with(std::vector<view_ptr>(), [&] (auto& views) {
        return parallel_for_each(result->rows().begin(), result->rows().end(), [&proxy, &views] (auto&& row) {
            return create_view_from_table_row(proxy, row).then([&views] (auto&& v) {
                views.push_back(std::move(v));
            });
        }).then([&views] {
            return std::move(views);
        });
    });
}
```

Which is correct.

Is there a core dump?

**Comment by amoskong** [comment]
No coredump

**Comment by duarten** [comment]
I think this was fixed by 4dfddc97c7888e534a5348af43a70d59a6572711. Has it happened again?

**Comment by duarten** [comment]
Closing as stale.
